ansible_collections/arista/avd/plugins/filter/password_filter.py

```diff
@@ -0,0 +1,84 @@
+"""
+Encrypt / Decrypt filters
+"""
+
+from ansible_collections.arista.avd.plugins.plugin_utils.bgp_utils import cbc_decrypt, cbc_encrypt
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import AristaAvdError, AristaAvdMissingVariableError
+
+
+##############
+# BGP
+##############
+def bgp_encrypt(password, key=None, usebase64=True) -> str:
+    """
+    Encrypt a password. The key is either <PEER_GROUP_NAME>_passwd or <NEIGHBOR_IP>_passwd
+
+    Returns the encrypted password as a string
+    """
+    if not key:
+        raise AristaAvdMissingVariableError("Key is required for BGP encryption")
+
+    data = bytes(password, encoding="utf-8")
+    key = bytes(f"{key}_passwd", encoding="utf-8")
+
+    return cbc_encrypt(key, data, usebase64).decode()
+
+
+def bgp_decrypt(password, key=None, usebase64=True) -> str:
```

Comment:
Why are you exposing but not documenting usebase64?

Reply:
I am not 100% sure that we should expose it - ideally we want our users to always use it (hence I did not document it). Thoughts?

Reply:
Removed all occurences in latest commit